opm/simulators/wells/WellBhpThpCalculator.cpp

```diff
@@ -839,6 +866,114 @@ bruteForceBracket(const std::function<double(const double)>& eq,
     return bracket_found;
 }
 
+bool WellBhpThpCalculator::
+isStableSolution(const WellState& well_state,
+                 const Well& well,
+                 const std::vector<double>& rates,
+                 const SummaryState& summaryState,
+                 DeferredLogger& deferred_logger) const
+{
+    assert(int(rates.size()) == 3); // the vfp related only supports three phases now.
+    assert(well_.isProducer()); // only valid for producers 
+
+    static constexpr int Gas = BlackoilPhases::Vapour;
+    static constexpr int Oil = BlackoilPhases::Liquid;
+    static constexpr int Water = BlackoilPhases::Aqua;
+
+    const double aqua = rates[Water];
+    const double liquid = rates[Oil];
+    const double vapour = rates[Gas];
+    const double thp = well_.getTHPConstraint(summaryState);
+
+    const auto& controls = well.productionControls(summaryState);
+    const auto& wfr =  well_.vfpProperties()->getExplicitWFR(controls.vfp_table_number, well_.indexOfWell());
+    const auto& gfr = well_.vfpProperties()->getExplicitGFR(controls.vfp_table_number, well_.indexOfWell());
+
+    const auto& table = well_.vfpProperties()->getProd()->getTable(controls.vfp_table_number);
+    const bool use_vfpexplicit = well_.useVfpExplicit();
+
+    detail::VFPEvaluation bhp = detail::bhp(table, aqua, liquid, vapour, thp, well_.getALQ(well_state), wfr, gfr, use_vfpexplicit);
+
+    bhp.value = bhp.value + getVfpBhpAdjustment(bhp.value, thp);
```

Comment:
True, removed. I think I originally left it there because I was a bit puzzled why `getVfpBhpAdjustment` only returns a `double` since it's a function of `bhp` and `thp` (and should therefore have non-zero derivatives). I have not looked further into this, however, so will just remove the line for now.